File: Array/Javascript/Question2/Solution1.js
let arr2 = [1, 2, 3, 4, 5];
// Swap elements in place using two pointers
let left = 0, right = arr2.length - 1;
while (left < right) {
  [arr2[left], arr2[right]] = [arr2[right], arr2[left]];
  left++;
  right--;
}
console.log("Reversed Array:", arr2);

File: Array/Java/Question2/Solution_1.java
public static void reverseArrayTwoPointer(int[] arr) {
    int left = 0;              // Start pointer
    int right = arr.length - 1; // End pointer

    while (left < right) {
        // Swap elements at left and right
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;

        // Move pointers closer
        left++;
        right--;
    }
}
